### PR TITLE
[react-native] Account for `nativeOnly` props in `requireNativeComponent`

### PR DESCRIPTION
**Pull request by ide**
Adjusted the prop types of the returned native component class so that (a) the `nativeOnly` set of props adds to the prop types and (b) not all the props from the JS component are required (using `Partial`).

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/native-components-ios#properties (search for "nativeOnly")
- [x] Increase the version number in the header if appropriate. **N/A**
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. **N/A**

### CODE REVIEW

**Comment by ide** [comment]
@timwangdev - I added a test case.

**Review by timwangdev — APPROVED** [review]
LGTM

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -8970,11 +8970,13 @@ export interface ComponentInterface<P> {
...
  */
-export function requireNativeComponent<P>(
+export function requireNativeComponent<P, NP = {}>(
     viewName: string,
     componentInterface?: ComponentInterface<P>,
-    extraConfig?: { nativeOnly?: any }
-): React.ComponentClass<PropTypes.InferProps<PropTypes.ValidationMap<P>>>;
+    extraConfig?: { nativeOnly?: NP }
+): React.ComponentClass<
+    Partial<PropTypes.InferProps<PropTypes.ValidationMap<P>>> & { [K in keyof NP]?: any}
```
Comment (alloy) [inline types/react-native/index.d.ts]: What is the last mapped typed addition for?

Comment (ide) [reply types/react-native/index.d.ts]: The "nativeOnly" map, which looks like `{ prop1: true, prop2: true, ...}`, specifies props that are defined by the native Obj-C/Java module but aren't declared in the JS component's `propTypes`. (Also, all these prop type declarations are going away in RN anyway so expect to see a PR that deletes my recent PRs :P)

**Pull request merged**